src/core/dependenciesMintPolicy.test.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+
+import deepFreeze from "deep-freeze";
+import {NodeAddress} from "./graph";
+import {
+  _alignPeriodsToIntervals,
+  processMintPolicy,
+} from "./dependenciesMintPolicy";
+
+describe("core/dependenciesMintPolicy", () => {
+  describe("_alignPeriodsToIntervals", () => {
+    it("handles a case with no periods and no intervals", () => {
+      expect(_alignPeriodsToIntervals([], [])).toEqual([]);
+    });
+    it("handles a case with no periods and intervals", () => {
+      expect(_alignPeriodsToIntervals([], [1, 2])).toEqual([0, 0]);
+    });
+    it("handles a case with a single period that spans all time", () => {
+      const period = {startTimeMs: -Infinity, weight: 0.5};
+      expect(_alignPeriodsToIntervals([period], [1, 2])).toEqual([0.5, 0.5]);
+    });
+    it("handles a case with a single period that starts midway", () => {
+      const period = {startTimeMs: 1, weight: 0.5};
+      expect(_alignPeriodsToIntervals([period], [0, 1, 2])).toEqual([
+        0,
+        0.5,
+        0.5,
+      ]);
+    });
+    it("handles a case with a multiple periods", () => {
+      const period1 = {startTimeMs: 10, weight: 0.5};
+      const period2 = {startTimeMs: 20, weight: 0.1};
+      expect(
+        _alignPeriodsToIntervals([period1, period2], [0, 5, 10, 15, 20, 25])
+      ).toEqual([0, 0, 0.5, 0.5, 0.1, 0.1]);
+    });
```

Comment:
Should add a case where the period startTimeMs is in between a time interval, and possibly another where there's multiple periods between one time interval and the next

Reply:
Done.